### ISSUE TITLE
[BUG] using-feature-columns.ipynb nb yields an error

### ISSUE THREAD

**Issue by rnyak** [NVIDIA/NVTabular#497]
**Describe the bug**
`using-feature-columns.ipynb` nb yields the following error:

```
model.fit(
    train_ds,
    callbacks=[KerasSequenceValidater(val_ds)],
    epochs=10
)

---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-38-b4e46994caae> in <module>
      2     train_ds,
      3     callbacks=[KerasSequenceValidater(val_ds)],
----> 4     epochs=10
      5 )

/opt/conda/envs/rapids/lib/python3.7/site-packages/tensorflow/python/keras/engine/training.py in fit(self, x, y, batch_size, epochs, verbose, callbacks, validation_split, validation_data, shuffle, class_weight, sample_weight, initial_epoch, steps_per_epoch, validation_steps, validation_batch_size, validation_freq, max_queue_size, workers, use_multiprocessing)
   1094           use_multiprocessing=use_multiprocessing,
   1095           model=self,
-> 1096           steps_per_execution=self._steps_per_execution)
   1097 
   1098       # Container that configures and calls `tf.keras.Callback`s.

/opt/conda/envs/rapids/lib/python3.7/site-packages/tensorflow/python/keras/engine/data_adapter.py in __init__(self, x, y, sample_weight, batch_size, steps_per_epoch, initial_epoch, epochs, shuffle, class_weight, max_queue_size, workers, use_multiprocessing, model, steps_per_execution)
   1110         use_multiprocessing=use_multiprocessing,
   1111         distribution_strategy=ds_context.get_strategy(),
-> 1112         model=model)
   1113 
   1114     strategy = ds_context.get_strategy()

/opt/conda/envs/rapids/lib/python3.7/site-packages/tensorflow/python/keras/engine/data_adapter.py in __init__(self, x, y, sample_weights, shuffle, workers, use_multiprocessing, max_queue_size, model, **kwargs)
    907         max_queue_size=max_queue_size,
    908         model=model,
--> 909         **kwargs)
    910 
    911   @staticmethod

/opt/conda/envs/rapids/lib/python3.7/site-packages/tensorflow/python/keras/engine/data_adapter.py in __init__(self, x, y, sample_weights, workers, use_multiprocessing, max_queue_size, model, **kwargs)
    777     # Since we have to know the dtype of the python generator when we build the
    778     # dataset, we have to look at a batch to infer the structure.
--> 779     peek, x = self._peek_and_restore(x)
    780     peek = self._standardize_batch(peek)
    781     peek = _process_tensorlike(peek)

/opt/conda/envs/rapids/lib/python3.7/site-packages/tensorflow/python/keras/engine/data_adapter.py in _peek_and_restore(x)
    911   @staticmethod
    912   def _peek_and_restore(x):
--> 913     return x[0], x
    914 
    915   def _handle_multiprocessing(self, x, workers, use_multiprocessing,

/nvtabular/nvtabular/loader/tensorflow.py in __getitem__(self, idx)
    249         """
    250         try:
--> 251             return DataLoader.__next__(self)
    252         except StopIteration:
    253             # TODO: I would like to do a check for idx == 0

/nvtabular/nvtabular/loader/backend.py in __next__(self)
    253 
    254     def __next__(self):
--> 255         return self._get_next_batch()
    256 
    257     def _fetch_chunk(self):

/nvtabular/nvtabular/loader/backend.py in _get_next_batch(self)
    280         # get the first chunks
    281         if self._batch_itr is None:
--> 282             self._fetch_chunk()
    283 
    284         # try to iterate through existing batches

/nvtabular/nvtabular/loader/backend.py in _fetch_chunk(self)
    259         if isinstance(chunks, Exception):
    260             self.stop()
--> 261             raise chunks
    262         self._batch_itr = iter(chunks)
    263 

/nvtabular/nvtabular/loader/backend.py in load_chunks(self, dev, dataloader)
    118 
    119                     if len(chunks) > 0:
--> 120                         chunks = dataloader.make_tensors(chunks, dataloader._use_nnz)
    121                         # put returns True if buffer is stopped before
    122                         # packet can be put in queue. Keeps us from

/nvtabular/nvtabular/loader/backend.py in make_tensors(self, gdf, use_nnz)
    312     def make_tensors(self, gdf, use_nnz=False):
    313         for workflow in self.workflows:
--> 314             gdf = workflow.apply_ops(gdf)
    315         split_idx = self._get_segment_lengths(len(gdf))
    316 

/nvtabular/nvtabular/workflow.py in apply_ops(self, gdf, start_phase, end_phase, writer, output_path, dtypes)
    615         for phase_index in range(start, end):
    616             start = time.time()
--> 617             gdf = self._run_trans_ops_for_phase(gdf, self.phases[phase_index])
    618             self.timings["preproc_apply"] += time.time() - start
    619             if phase_index == len(self.phases) - 1 and writer and output_path:

/nvtabular/nvtabular/workflow.py in _run_trans_ops_for_phase(self, gdf, tasks)
    597                 gdf = op.apply_op(gdf, self.columns_ctx, cols_grp, target_cols, self.stats)
    598             elif isinstance(op, TransformOperator):
--> 599                 gdf = op.apply_op(gdf, self.columns_ctx, cols_grp, target_cols=target_cols)
    600         return gdf
    601 

/nvtabular/nvtabular/ops/transform_operator.py in apply_op(self, gdf, columns_ctx, input_cols, target_cols, stats_context)
     87     ):
     88         target_columns = self.get_columns(columns_ctx, input_cols, target_cols)
---> 89         new_gdf = self.op_logic(gdf, target_columns, stats_context=stats_context)
     90         self.update_columns_ctx(columns_ctx, input_cols, new_gdf.columns, target_columns)
     91         return self.assemble_new_df(gdf, new_gdf, target_columns)

/opt/conda/envs/rapids/lib/python3.7/contextlib.py in inner(*args, **kwds)
     72         def inner(*args, **kwds):
     73             with self._recreate_cm():
---> 74                 return func(*args, **kwds)
     75         return inner
     76 

/nvtabular/nvtabular/ops/hashed_cross.py in op_logic(self, gdf, target_columns, stats_context)
     52             val = 0
     53             for column in columns:
---> 54                 val ^= gdf[column].hash_values()  # or however we want to do this aggregation
     55             val = val % bucket_size
     56             new_gdf["_X_".join(columns)] = val

/opt/conda/envs/rapids/lib/python3.7/contextlib.py in inner(*args, **kwds)
     72         def inner(*args, **kwds):
     73             with self._recreate_cm():
---> 74                 return func(*args, **kwds)
     75         return inner
     76 

/opt/conda/envs/rapids/lib/python3.7/site-packages/cudf/core/dataframe.py in __getitem__(self, arg)
    651         """
    652         if is_scalar(arg) or isinstance(arg, tuple):
--> 653             return self._get_columns_by_label(arg, downcast=True)
    654 
    655         elif isinstance(arg, slice):

/opt/conda/envs/rapids/lib/python3.7/site-packages/cudf/core/frame.py in _get_columns_by_label(self, labels, downcast)
    441         If downcast is True, try and downcast from a DataFrame to a Series
    442         """
--> 443         new_data = self._data.select_by_label(labels)
    444         if downcast:
    445             if is_scalar(labels):

/opt/conda/envs/rapids/lib/python3.7/site-packages/cudf/core/column_accessor.py in select_by_label(self, key)
    217                 if any(isinstance(k, slice) for k in key):
    218                     return self._select_by_label_with_wildcard(key)
--> 219             return self._select_by_label_grouped(key)
    220 
    221     def select_by_index(self, index):

/opt/conda/envs/rapids/lib/python3.7/site-packages/cudf/core/column_accessor.py in _select_by_label_grouped(self, key)
    265 
    266     def _select_by_label_grouped(self, key):
--> 267         result = self._grouped_data[key]
    268         if isinstance(result, cudf.core.column.ColumnBase):
    269             return self.__class__({key: result})

KeyError: 'Age_Bucketize'
```
**Steps/Code to reproduce bug**
To reproduce the error, please run https://github.com/NVIDIA/NVTabular/blob/main/examples/tensorflow/using-feature-columns.ipynb. 

**Environment details (please complete the following information):**
 - Environment location: Docker on NGC
 - Method of NVTabular install: Docker, using `nvidia/nvtabular:0.3` image.

 TF version: `2.5.0-dev20201121`

**Additional context**

**Comment by rnyak** [comment]
@benfred the bug is due to naming in `new_col = f"{col}_{self._id}"` in bucketize op in old api.  The `new_col` name is generated as`Age_Bucketize_1` whereas model is expecting `Age_Bucketize`.

**Comment by benfred** [comment]
Fixed by #491
